### PR TITLE
[CUDA] Cleanup per-process-memory-fraction in `test_cuda.py` tests

### PR DESCRIPTION
**Pull request by eqy**
Otherwise certain sequences of tests will fail with OOM e.g.,
```
# python test/test_cuda.py -k max_split_expandable -k test_assigning_back_deleter_fns_to_tensor  --repeat 100                                                                                                                                                                                                                                                                                          ..                                                                                                                                                                                                                                                                                                                                                                                                                                         ----------------------------------------------------------------------                                                                                                                                                                                                                                                                                                                                                                     Ran 2 tests in 0.311s                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 OK                                                                                                                                                                                                                                                                                                                                                                                                                                         E.                                                                                                                                                                                                                                                                                                                                                                                                                                         ======================================================================                                                                                                                                                                                                                                                                                                                                                                     ERROR: test_assigning_back_deleter_fns_to_tensor (__main__.TestBlockStateAbsorption.test_assigning_back_deleter_fns_to_tensor)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/pytorch/torch/testing/_internal/common_utils.py", line 3058, in wrapper
    method(*args, **kwargs)
  File "/workspace/pytorch/test/test_cuda.py", line 4320, in test_assigning_back_deleter_fns_to_tensor
    graph, outputs = cudagraphify(foo, [inp])
                     ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/workspace/pytorch/test/test_cuda.py", line 4080, in cudagraphify
    fn(*inputs)
  File "/workspace/pytorch/test/test_cuda.py", line 4316, in foo
    int8_cuda(LARGE_BUFFER) + x,
    ~~~~~~~~~~~~~~~~~~~~~~~~^~~
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 160.00 MiB. GPU 0 has a total capacity of 31.73 GiB of which 31.30 GiB is free. Process 2916661 has 442.00 MiB memory in use. 120.00 MiB allowed; Of the allocated memory 52.00 MiB is allocated by PyTorch, and 6.00 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)

To execute this test, run the following from the base repo dir:
    python test/test_cuda.py TestBlockStateAbsorption.test_assigning_back_deleter_fns_to_tensor                                                                                                                                                                                                                                                                                                                                            
This message can be suppressed by setting PYTORCH_PRINT_REPRO_ON_FAILURE=0

----------------------------------------------------------------------
Ran 2 tests in 0.136s

FAILED (errors=1)
```

cc @ptrblck @msaroufim

### CODE REVIEW

File: test/test_cuda.py [hunk 1]
```diff
@@ -322,6 +322,7 @@ def test_set_per_process_memory_fraction(self):
         # ensure out of memory error doesn't disturb subsequent kernel
         tensor.fill_(1)
         self.assertTrue((tensor == 1).all())
+        torch.cuda.set_per_process_memory_fraction(1.0, 0)
```
Comment (Skylion007) [inline test/test_cuda.py:325]: We should create a context manager / fixture for this going forward...

**Review by Skylion007 — APPROVED** [review]

File: test/test_cuda.py [hunk 2]
```diff
@@ -3642,6 +3643,7 @@ def alloc(n):
         del a, b, c
         # force release_cached_blocks to run with some expandable segments in the free list
         alloc(120)
+        torch.cuda.memory.set_per_process_memory_fraction(1.0)
```
Comment (Skylion007) [inline test/test_cuda.py:3646]: What if the test fails? Then the memory fraction will never be set and subsequent tests will fail too?

File: test/test_cuda.py (hunk 1 above)
Comment (eqy) [reply test/test_cuda.py:325]: Yeah this is pretty annoying/involved atm because it requires e.g., changing the PluggableAllocator interface

**Pull request closed without merging**